include/nanobind/nb_cast.h

```diff
@@ -123,7 +124,9 @@ NB_INLINE uint8_t flags_for_local_caster(uint8_t flags) noexcept {
            into storage owned by the caster, which won't live long enough.
            Exception: the 'char' caster produces a result that points to
            storage owned by the incoming Python 'str' object, so it's OK. */
-        static_assert(!is_ref || std::is_same_v<T, const char*>,
+        static_assert(!is_ref || std::is_same_v<T, const char*> ||
+                      std::conjunction_v<std::is_pointer<T>,
```

Comment:
Can you replace this with ordinary constexpr boolean expressions? (&&). Conjunction_v is unnecessarily heavy.